src/dataset/nexarear.py

```diff
@@ -0,0 +1,322 @@
+      global_step: step of the checkpoint
+      all_boxes: all_boxes[cls][image] = N x 5 arrays of 
+        [xmin, ymin, xmax, ymax, score]
+    Returns:
+      aps: array of average precisions.
+      names: class names corresponding to each ap
+    """
+    det_file_dir = os.path.join(
+        eval_dir, 'detection_files_{:s}'.format(global_step), 'data')
+    if not os.path.isdir(det_file_dir):
+      os.makedirs(det_file_dir)
+
+    for im_idx, index in enumerate(self._image_idx):
+      filename = os.path.join(det_file_dir, index+'.txt')
+      with open(filename, 'wt') as f:
+        for cls_idx, cls in enumerate(self._classes):
+          dets = all_boxes[cls_idx][im_idx]
+          for k in xrange(len(dets)):
+            f.write(
+                '{:s} -1 -1 0.0 {:.2f} {:.2f} {:.2f} {:.2f} 0.0 0.0 0.0 0.0 0.0 '
+                '0.0 0.0 {:.3f}\n'.format(
+                    cls.lower(), dets[k][0], dets[k][1], dets[k][2], dets[k][3],
+                    dets[k][4])
+            )
+
+    cmd = self._eval_tool + ' ' \
+          + os.path.join(self._data_root_path, 'training') + ' ' \
+          + os.path.join(self._data_root_path, 'ImageSets',
+                         self._image_set+'.txt') + ' ' \
+          + os.path.dirname(det_file_dir) + ' ' + str(len(self._image_idx))
+
+    print('Running: {}'.format(cmd))
+    status = subprocess.call(cmd, shell=True)
+
+    aps = []
+    names = []
+    for cls in self._classes:
+      det_file_name = os.path.join(
+          os.path.dirname(det_file_dir), 'stats_{:s}_ap.txt'.format(cls))
+      if os.path.exists(det_file_name):
+        with open(det_file_name, 'r') as f:
+          lines = f.readlines()
+        assert len(lines) == 3, \
+            'Line number of {} should be 3'.format(det_file_name)
+
+        aps.append(float(lines[0].split('=')[1].strip()))
+        aps.append(float(lines[1].split('=')[1].strip()))
+        aps.append(float(lines[2].split('=')[1].strip()))
+      else:
+        aps.extend([0.0, 0.0, 0.0])
+
+      names.append(cls+'_easy')
+      names.append(cls+'_medium')
+      names.append(cls+'_hard')
+
+    return aps, names
+
+  def do_detection_analysis_in_eval(self, eval_dir, global_step):
+    det_file_dir = os.path.join(
+        eval_dir, 'detection_files_{:s}'.format(global_step), 'data')
+    det_error_dir = os.path.join(
+        eval_dir, 'detection_files_{:s}'.format(global_step),
+        'error_analysis')
+    if not os.path.exists(det_error_dir):
+      os.makedirs(det_error_dir)
+    det_error_file = os.path.join(det_error_dir, 'det_error_file.txt')
+
+    stats = self.analyze_detections(det_file_dir, det_error_file)
+    ims = self.visualize_detections(
+        image_dir=self._image_path,
+        image_format='.png',
+        det_error_file=det_error_file,
+        output_image_dir=det_error_dir,
+        num_det_per_type=10
+    )
+
+    return stats, ims
+
+  def analyze_detections(self, detection_file_dir, det_error_file):
```

Comment:
this code is duplicated from kitti, no need for this code duplication when using @orenmeiri benchmark code